pkg/machines/components/vmSnapshotsCreateModal.jsx

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import cockpit from "cockpit";
+import React from "react";
+import moment from "moment";
+
+import { Modal, FormGroup, HelpBlock } from "patternfly-react";
+import {
+    Button,
+    TextArea,
+    TextInput
+} from "@patternfly/react-core";
+
+import { ModalError } from "cockpit-components-inline-notification.jsx";
+import { createSnapshot } from "../libvirt-dbus.js";
+import { getVmSnapshots } from '../actions/provider-actions.js';
+
+const _ = cockpit.gettext;
+
+const NameRow = ({ onValueChanged, dialogValues, onValidate }) => {
+    return (
+        <>
+            <label className="control-label" htmlFor="name">
+                {_("Name")}
+            </label>
+            <FormGroup validationState={dialogValues.validationError.name ? "error" : undefined} controlId="name">
+                <TextInput value={dialogValues.name}
+                    id="name"
+                    type="text"
+                    onChange={(value) => onValueChanged("name", value)}
+                    onBlur={() => onValidate("name")}
+                    aria-label={_("Name input text")}
+                />
+                { dialogValues.validationError.name &&
+                <HelpBlock>
+                    <p className="text-danger">{dialogValues.validationError.name}</p>
+                </HelpBlock> }
+            </FormGroup>
+
+        </>
+    );
+};
+
+const DescriptionRow = ({ onValueChanged, dialogValues }) => {
+    return (
+        <>
+            <label className="control-label" htmlFor="description">
+                {_("Description")}
+            </label>
+            <TextArea value={dialogValues.description}
+                id="description"
+                placeholder={_("Optional arbitrary description of a snapshot")}
```

Comment:
This is very complex wording. _Description_ is already mentioned in the label, and Snapshot is clear from the context.
I would suggest simply _Optional_